packages/scm/src/browser/scm-widget.tsx

```diff
@@ -154,540 +133,45 @@ export class ScmWidget extends ReactWidget implements StatefulWidget {
-            if (widget) {
-                const diffNavigator = this.diffNavigatorProvider(widget.editor);
-                if (diffNavigator.canNavigate() && diffNavigator.hasNext()) {
-                    diffNavigator.next();
-                } else {
-                    const next = repository.selectNextResource();
-                    if (next) {
-                        next.open();
-                    }
-                }
-            } else {
-                const next = repository.selectNextResource();
-                if (next) {
-                    next.open();
-                }
-            }
-        } else if (repository && repository.resources.length) {
-            repository.selectedResource = repository.resources[0];
-            repository.selectedResource.open();
-        }
-    }
-
-    protected async openResource(resource: ScmResource): Promise<EditorWidget | undefined> {
-        try {
-            await resource.open();
-        } catch (e) {
-            console.error('Failed to open a SCM resource', e);
-            return undefined;
-        }
-
-        let standaloneEditor: EditorWidget | undefined;
-        const resourcePath = resource.sourceUri.path.toString();
-        for (const widget of this.editorManager.all) {
-            const resourceUri = widget.getResourceUri();
-            const editorResourcePath = resourceUri && resourceUri.path.toString();
-            if (resourcePath === editorResourcePath) {
-                if (widget.editor.uri.scheme === DiffUris.DIFF_SCHEME) {
-                    // prefer diff editor
-                    return widget;
-                } else {
-                    standaloneEditor = widget;
-                }
-            }
-            if (widget.editor.uri.scheme === DiffUris.DIFF_SCHEME
-                && String(widget.getResourceUri()) === resource.sourceUri.toString()) {
-                return widget;
-            }
-        }
-        // fallback to standalone editor
-        return standaloneEditor;
-    }
-
-    protected selectPreviousResource(): ScmResource | undefined {
-        const repository = this.scmService.selectedRepository;
-        return repository && repository.selectPreviousResource();
-    }
-
-    protected selectNextResource(): ScmResource | undefined {
-        const repository = this.scmService.selectedRepository;
-        return repository && repository.selectNextResource();
-    }
-
-    protected openSelected(): void {
-        const repository = this.scmService.selectedRepository;
-        const resource = repository && repository.selectedResource;
-        if (resource) {
-            resource.open();
-        }
+        this.commitWidget.focus();
     }
 
     storeState(): any {
-        const repository = this.scmService.selectedRepository;
-        return repository && repository.input;
+        console.warn('saving state');
+        const state: object = {
+            commitState: this.commitWidget.storeState(),
```

Comment:
you don't need to call store/restore state on widget, the widget layout restorer will take care about it if such widgets are created via the widget factory and implement `StatefulWidget` interace.

```ts
return {
   commit: this.commitWidget,
   changes: this.resourceWidget
}
```

should work

Reply:
I could not get this to work without making it all more complicated.

Reply:
ok, let's go with it like now

> all more complicated.

You mean because commit and resource widget should be created via the factory?